src/main/java/net/minecraftforge/common/plants/IGrowablePlant.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2018.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.plants;
+
+import java.util.Random;
+
+import net.minecraft.block.IGrowable;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+
+/**
+ * The base interface of all growable plants. Implementers must override canGrow/grow, or callers will end up with a StackOverflowException. This class extends IGrowable in order
+ * to reduce the number of patches needed. Modders should not override the IGrowable methods.
+ */
+@SuppressWarnings("deprecation")
+public interface IGrowablePlant extends IPlant, IGrowable
```

Comment:
Why not make IGrowable extend this interface? Seems like it'd save a lot of patches and not need the 'modders should not override the IGrowable methods'